src/transport/MessageCounter.h

```diff
@@ -88,44 +88,44 @@ class GlobalEncryptedMessageCounter : public MessageCounter
 #else
     struct FakePersistedCounter
     {
-        FakePersistedCounter() : value(0) {}
+        FakePersistedCounter() : mValue(0) {}
         CHIP_ERROR Init(chip::Platform::PersistedStorage::Key aId, uint32_t aEpoch) { return CHIP_NO_ERROR; }
 
-        uint32_t GetValue() { return value; }
+        uint32_t GetValue() { return mValue; }
         CHIP_ERROR Advance()
         {
-            ++value;
+            ++mValue;
             return CHIP_NO_ERROR;
         }
 
     private:
-        uint32_t value;
+        uint32_t mValue;
     } persisted;
 #endif
 };
 
 class LocalSessionMessageCounter : public MessageCounter
 {
 public:
-    static constexpr uint32_t kInitialValue                 = 1;         ///< Used for initializing peer counter
+    static constexpr uint32_t kInitialSyncValue             = 0;         ///< Used for initializing peer counter
     static constexpr uint32_t kMessageCounterRandomInitMask = 0x0FFFFFF; ///< 28-bit mask
 
     /**
      * Initialize a local message counter with random value between [0, 2^28-1]. This increases the difficulty of traffic analysis
      * attacks by making it harder to determine how long a particular session has been open.
      */
-    LocalSessionMessageCounter() { value = Crypto::GetRandU32() & kMessageCounterRandomInitMask; }
+    LocalSessionMessageCounter() { mValue = (Crypto::GetRandU32() + 1) & kMessageCounterRandomInitMask; }
```

Comment:
Don't `GetRandU32()` and `GetRandU32() + 1` have  an identical distribution given that we mask them (i.e. they will effectively wrap over)?

I do not understand this change and it makes the code more complex. If this is actually intended (I am really unclear how that could be though ... chances of a 0 should be the same as for a 0xFFFFFF) please add a comment.

Reply:
I tried comparing something like this (much smaller bin size to actually see results):

```py
import matplotlib.pyplot as plt
import random
%matplotlib inline

plt.hist([random.randint(0, 0xFFFFFFFF) & 0xF for x in range(200000)], bins=range(0x0F + 2))
plt.hist([(random.randint(0, 0xFFFFFFFF) + 1) & 0xF for x in range(200000)], bins=range(0x0F + 2))
```

they are the same (except slight differences due to random ... for sure no difference around 0 or 0x0F

Reply:
My mistake, need to increment after mask, will fix.